psl-parser/src/main/java/org/linqs/psl/parser/CommandLineLoader.java

```diff
@@ -0,0 +1,421 @@
+                .desc("Directly specify PSL properties (overrides options set via --" + OPTION_PROPERTIES_FILE_LONG + ")." +
+                        " See https://github.com/linqs/psl/wiki/Configuration-Options for a list of available options." +
+                        " Log4j properties (properties starting with 'log4j') will be passed to the logger." +
+                        " 'log4j.threshold=DEBUG', for example, will be passed to log4j and set the global logging threshold.")
+                .hasArg()
+                .numberOfArgs(2)
+                .valueSeparator('=')
+                .build());
+        return options;
+
+    }
+
+    private static HelpFormatter getHelpFormatter() {
+        HelpFormatter helpFormatter = new HelpFormatter();
+
+        // Hack the option ordering to put argumentions without options first and then required options first.
+        // infer and learn go first, then required, then just normal.
+        helpFormatter.setOptionComparator(new Comparator<Option>() {
+            @Override
+            public int compare(Option o1, Option o2) {
+                String name1 = o1.getOpt();
+                if (name1 == null) {
+                    name1 = o1.getLongOpt();
+                }
+
+                String name2 = o2.getOpt();
+                if (name2 == null) {
+                    name2 = o2.getLongOpt();
+                }
+
+                if (name1.equals(OPERATION_INFER)) {
+                    return -1;
+                }
+
+                if (name2.equals(OPERATION_INFER)) {
+                    return 1;
+                }
+
+                if (name1.equals(OPERATION_LEARN)) {
+                    return -1;
+                }
+
+                if (name2.equals(OPERATION_LEARN)) {
+                    return 1;
+                }
+
+                if (o1.isRequired() && !o2.isRequired()) {
+                    return -1;
+                }
+
+                if (!o1.isRequired() && o2.isRequired()) {
+                    return 1;
+                }
+
+                return name1.compareTo(name2);
+            }
+        });
+
+        helpFormatter.setWidth(100);
+
+        return helpFormatter;
+    }
+
+    /**
+     * Parse the options on the command line.
+     * Will exit on error, but Will return null if the CLI should not be run (like if we are doing a help/version run).
+     */
+    private static CommandLine parseOptions(String[] args) {
+        Options options = setupOptions();
+        CommandLineParser parser = new DefaultParser();
+        CommandLine commandLineOptions = null;
+
+        try {
+            commandLineOptions = parser.parse(options, args);
+        } catch (ParseException ex) {
+            System.err.println("Command line error: " + ex.getMessage());
+            getHelpFormatter().printHelp("psl", options, true);
+            System.exit(1);
+        }
+        if (commandLineOptions.hasOption(OPTION_HELP)) {
```

Comment:
There should be a newline between all these `if`s.
Use vertical whitespace to separate code that is cognitively different.
(We don't need to think about both `if`s at the same time.)